src/components/Badge/Badge.vue

```diff
@@ -0,0 +1,58 @@
+<template>
+  <div class="inline-block">
+    <div
+      :class="['flex justify-center align-center',
+               { 'px-4 py-2 rounded-full': rounded },
+               { 'px-2 py-1 rounded-sm': square },
+               { 'bg-white-300 text-gray-500 hover:bg-gray-100': info },
+               { 'bg-mint-100 text-mint-900 hover:bg-mint-300': success && !info },
+               { 'bg-lemon-100 text-lemon-900 hover:bg-lemon-300': warning && !info && !success },
+               { 'bg-coral-100 text-coral-900 hover:bg-coral-300': error && !info && !success && !warning },
+               { 'bg-turquoise-100 text-turquoise-900 hover:bg-turquoise-300': error && !info && !success && !warning && !error },
+               { 'bg-flint-100 text-gray-700 hover:bg-flint-300': error && !info && !success && !warning && !error && !turquoise }
```

Comment:
I hear you. I prefer using a component like this though 
```
<badge success>Success message here</badge>
```

And boolean props give you that. Then you don't have magic string floating around.

But this may be something for us to make a team decision about (how to handle variants) because there are a lot of them across our components.

Reply:
I do like boolean props too and for the same reason, and generally prefer them over 'magic strings'. They just look nicer.

However in these type of cases like variants, the string names are pretty standard, and you would avoid:
1. Unpredictable: `<badge success warning>??</badge>
2. Boolean logic: `error && !info && !success && !warning && !error && !turquoise`

That's why in cases like these I prefer just one prop. I think it can prevent certain errors. But as you say, let's discuss it with the team so we arrive to a standard solution.